maufbapi/mqtt/conn.py

```diff
@@ -384,6 +384,17 @@ async def _post_connect(self) -> None:
         else:
             await self.publish(RealtimeTopic.SYNC_CREATE_QUEUE, self._sync_create_queue_data)
 
+    async def _post_connect(self) -> None:
+        try:
+            await self._unsafe_post_connect()
+        except Exception:
+            # If we ever connect, but fail to send the SYNC_* message, we end up stuck with a "working"
+            # MQTT connection but no data flowing. Always retry in this situation. The listen method
+            # should detect & raise any connection issues, so looping here is OK.
+            self.log.exception("Error publishing MQTT queue SYNC request, retrying in 5s!")
+            await asyncio.sleep(5)
+            background_task.create(self._post_connect())
```

Comment:
Creating a recursive background task for this is slightly sus. Might be better to have an explicit loop and cancel the whole `_post_connect` task in `_on_disconnect_handler` to ensure it doesn't get stuck

Reply:
https://github.com/mautrix/facebook/pull/304/commits/eba08bffee6cbf8e3d88a7890471cbeb68db69bb